packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesController.kt

```diff
@@ -44,18 +50,97 @@ class UpdatesController private constructor(
 
   private var launcher: Launcher? = null
   val databaseHolder = DatabaseHolder(UpdatesDatabase.getInstance(context))
+  // TODO: move away from DatabaseHolder pattern to Handler thread
```

Comment:
Curious also about the option of a coroutine pattern.

Coroutines are lighter so seem preferable unless there is a reason not to?

Reply:
That's a great point, I didn't think of coroutines. Will look into it...

Reply:
Have looked into this a bit -- in order to replace the lock mechanism and replicate the GCD behavior from iOS, we need coroutines to execute serially and in a predictable order. It looks like this is possible (using channels and a custom coroutine executor) but not exactly what coroutines are intended for. I think it might still be a nice/worthwhile change but would like to explore this in a separate PR, since I know this current implementation works.